addons/equipviewer/icon_extractor.lua

```diff
@@ -0,0 +1,255 @@
+local icon_extractor = {}
+
+local game_path = windower.pol_path..'\/..\/FINAL FANTASY XI'
+
+local string = require('string')
+local io = require('io')
+local math = require('math')
+
+local concat = table.concat
+local floor = math.floor
+local byte = string.byte
+local char = string.char
+local sub = string.sub
+
+local file_size = '\122\16\00\00'
+local reserved1 = '\00\00'
+local reserved2 = '\00\00'
+local starting_address = '\122\00\00\00'
+
+local default = '\00\00\00\00'
+
+local dib_header_size = '\108\00\00\00'
+local bitmap_width = '\32\00\00\00'
+local bitmap_height = '\32\00\00\00'
+local n_color_planes = '\01\00'
+local bits_per_pixel = '\32\00'
+local compression_type = '\03\00\00\00'
+local image_size = '\00\16\00\00'
+local h_resolution_target = default
+local v_resolution_target = default
+local default_n_colors = default
+local important_colors = default
+local alpha_mask = '\00\00\00\255'
+local red_mask = '\00\00\255\00'
+local green_mask = '\00\255\00\00'
+local blue_mask = '\255\00\00\00'
+local colorspace = 'sRGB'
+local endpoints = string.rep('\00', 36)
+local red_gamma = default
+local green_gamma = default
+local blue_gamma = default
+
+local header = 'BM' .. file_size .. reserved1 .. reserved2 .. starting_address
+        .. dib_header_size .. bitmap_width .. bitmap_height .. n_color_planes
+        .. bits_per_pixel .. compression_type .. image_size
+        .. h_resolution_target .. v_resolution_target
+        .. default_n_colors .. important_colors
+        .. red_mask .. green_mask .. blue_mask .. alpha_mask
+        .. colorspace .. endpoints .. red_gamma .. green_gamma .. blue_gamma
+
+--local icon_file = io.open('C:/Program Files (x86)/PlayOnline/SquareEnix/FINAL FANTASY XI/ROM/118/106.DAT', 'rb')
+local color_lookup = {}
+local bmp_segments = {}
+
+for i = 0, 255 do
+    color_lookup[string.char(i)] = ''
+end
+
+--[[
+3072 bytes per icon
+640 bytes for stats, string table, etc.
+2432 bytes for pixel data
+--]]
+
+local item_dat_map = {
+    [1]={min=1, max=4095, dat_path='118/106', offset=-1},
+    [2]={min=4096, max=8191, dat_path='118/107', offset=0},
+    [3]={min=8192, max=8703, dat_path='118/110', offset=0},
+    [4]={min=8704, max=10239, dat_path='301/115', offset=0},
+    [5]={min=10240, max=16383, dat_path='118/109', offset=0},
+    [6]={min=16384, max=23039, dat_path='118/108', offset=0},
+    [7]={min=23040, max=28671, dat_path='286/73', offset=0},
+    [8]={min=28672, max=29695, dat_path='217/21', offset=0},
+    [9]={min=29696, max=30719, dat_path='288/80', offset=0},
+    [10]={min=61440, max=61951, dat_path='288/67', offset=0},
+    [11]={min=65535, max=65535, dat_path='174/48', offset=0},
+}
+
+local item_by_id = function (id, output_path)
+    local dat_stats = find_item_dat_map(id)
+    local icon_file = open_dat(dat_stats)
+    
+    local id_offset = dat_stats.min + dat_stats.offset
+    icon_file:seek('set', (id - id_offset) * 0xC00 + 0x2BD)
+    local data = icon_file:read(2048)
```

Comment:
This will sound like a super petty complaint, but it kinda irks me you use hex above but decimal here 😅 `0x800` would look better here imo.

Reply:
Would you like me to just change most of the numbers to hex? Or just these two?

Reply:
Personally I think hex should be used everywhere where it is relevant. The size you're reading here is based on the hexadecimal system (same as the base item memory size with `0xC00`), so I think it makes sense. Honestly, in this file most numbers would make sense as hex values, but probably not all. The ones I commented on where just two examples where it stood out to me, but it would need to be looked at on a case-by-case basis. Either way, I wouldn't invest too much time into this, was just a suggestion in the first place.